### ISSUE TITLE
GitHub action

### ISSUE THREAD

**Issue by Veetaha** [est31/cargo-udeps#68]
Would there be an interest to roll out a GitHub action for `cargo-udeps` so that adding this one to CI runs would be just a one-liner?
There is an awesome template by embark guys with their `cargo-deny` sanitizer we could use as a reference, which is nothing more than a Dockerfie and 3-4 lines of bash at it's core ([link](https://github.com/EmbarkStudios/cargo-deny-action))
There is also an [opened issue](https://github.com/EmbarkStudios/cargo-deny/issues/74) to think about somehow merging both together, but this is yet another topic :D

The action will probably reside in a separate repo (since it involves tagging versions of the action and this may interfere with `cargo-udeps` cli releases).

**Comment by est31** [comment]
This is an interesting idea.

**Comment by S-Coyle** [comment]
Was just looking for exactly this, +1 from me

**Comment by aig787** [comment]
I needed this so I took a stab at it, borrowing heavily from actions-rs. https://github.com/aig787/cargo-udeps-action. I'm happy to have it brought into the family if you want @est31.

**Comment by tisonkun** [comment]
@est31 I build a simple workflow for running cargo-udeps in GitHub Actions. It wouldn't have to build a dedicated action IMO.

https://github.com/engula/engula/runs/4332111979?check_suite_focus=true

However, installing cargo-udeps takes about 5 minutes. Could we reduce the overhead?

**Comment by mati865** [comment]
You could reduce it by caching built binary.

**Comment by est31** [comment]
Yeah there is a [Rust specific section](https://github.com/actions/cache/blob/main/examples.md#rust---cargo) in the github actions cache documentation.
